core/src/test/java/com/graphhopper/routing/ev/EnumEncodedValueTest.java

```diff
@@ -32,4 +41,44 @@ public void testSize() {
         assertEquals(4, 32 - Integer.numberOfLeadingZeros(16 - 1));
         assertEquals(5, 32 - Integer.numberOfLeadingZeros(17 - 1));
     }
+
+    @Test
+    public void serializationAndDeserialization() throws JsonProcessingException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+
+        List<EncodedValue> encodedValues = new ArrayList<>();
+        // add enum, int, decimal and boolean encoded values
+        encodedValues.add(new EnumEncodedValue<>(RoadClass.KEY, RoadClass.class));
+        encodedValues.add(Lanes.create());
+        encodedValues.add(MaxWidth.create());
+        encodedValues.add(GetOffBike.create());
+
+        // serialize
+        List<String> serializedEVs = new ArrayList<>();
+        for (EncodedValue e : encodedValues) {
+            String s = mapper.writeValueAsString(e);
+            serializedEVs.add(s);
+        }
+
+        // deserialize
+        List<EncodedValue> deserializedEVs = new ArrayList<>();
+        for (String s : serializedEVs) {
+            JsonNode jsonNode = mapper.readTree(s);
+            deserializedEVs.add(mapper.treeToValue(jsonNode, EncodedValue.class));
+        }
```

Comment:
This is pretty nice and seems to work for all our `EncodedValue` subclasses automatically, due to the `@JsonTypeInfo` annotation that I added to the `EncodedValue` interface.